cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -451,16 +582,28 @@ func (ht *hcsTask) Wait() *task.StateResponse {
 	return ht.init.Wait()
 }
 
-func (ht *hcsTask) waitInitExit() {
+func (ht *hcsTask) waitInitExit(destroyContainer bool) {
 	ctx, span := trace.StartSpan(context.Background(), "hcsTask::waitInitExit")
 	defer span.End()
 	span.AddAttributes(trace.StringAttribute("tid", ht.id))
 
 	// Wait for it to exit on its own
 	ht.init.Wait()
 
-	// Close the host and event the exit
-	ht.close(ctx)
+	if destroyContainer {
+		// Close the host and event the exit
+		ht.close(ctx)
+	} else {
+		// Close the container's host, but do not close or terminate the container itself
+		ht.closeHost(ctx)
```

Comment:
If we close the host (UVM) here, doesn't that interfere with our ability to save it as a template?

Reply:
This works because for WCOW the task which owns the host is not the `hcsTask` instead it is the `wcowPodSandboxTask` and so the `closeHost` method doesn't actually shutdown the uvm. I can still add a check here to not call `closeHost` at all if we are planning to save it as a template.

Reply:
It looks to me like `ht` is always a `hcsTask` here, and there is no `closeHost` on `wcowPodSandboxTask`. That seems like we will always end up calling `(hcsTask).closeHost`. Maybe this is fine because presumably `ht.ownsHost` will be `false` for the workload container, so calling `closeHost` won't actually do anything other than send a task exit event for the container?

Reply:
That's correct. That's exactly why we can safely save the uvm even after calling `closeHost`.

Reply:
Okay, that makes sense. The comment here could be clarified, I think It currently sounds to me like "closing the container's host" would mean closing the UVM.